phd-tests/framework/src/disk/file.rs

```diff
@@ -31,23 +93,19 @@ impl FileBackedDisk {
     /// the specified artifact on the host file system.
     pub(crate) fn new_from_artifact(
         backend_name: String,
-        artifact_path: &impl AsRef<Path>,
-        data_dir: &impl AsRef<Path>,
+        artifact_path: &impl AsRef<Utf8Path>,
+        data_dir: &impl AsRef<Utf8Path>,
         guest_os: Option<GuestOsKind>,
     ) -> Result<Self, super::DiskError> {
-        let mut disk_path = data_dir.as_ref().to_path_buf();
-        disk_path.push(format!("{}.phd_disk", Uuid::new_v4()));
-        info!(
-            source = %artifact_path.as_ref().display(),
-            disk_path = %disk_path.display(),
-            "Copying source image to create temporary disk",
-        );
-
-        std::fs::copy(artifact_path, &disk_path)?;
+        let artifact = BackingFile::create_from_source(
+            artifact_path.as_ref(),
+            data_dir.as_ref(),
+        )?;
+        let disk_path = artifact.path().into_std_path_buf();
 
         // Make sure the disk is writable (the artifact may have been
         // read-only).
-        let disk_file = std::fs::File::open(&disk_path)?;
+        let disk_file = std::fs::File::open(disk_path)?;
```

Comment:
nit: `File::open` takes `impl AsRef<Path>`, so I don't _think_ the explicit conversion to a `std::path::PathBuf` is actually necessary?

Reply:
Good call. Done in c2133469b.